packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts

```diff
@@ -81,7 +104,7 @@ export class BaseTypesVisitor<TRawConfig extends RawTypesConfig = RawTypesConfig
 
     return new DeclarationBlock(this._declarationBlockConfig)
       .export()
-      .asKind('type')
+      .asKind(this._parsedConfig.declarationKind.scalar)
```

Comment:
This is the gist of the PR. Use config for the output type rather than a static kind.